From c2adec2b09f37a6f8cf43f3344f50a4b2b8e310a Mon Sep 17 00:00:00 2001
From: John Howard <howardjohn@google.com>
Date: Tue, 9 Nov 2021 12:46:49 -0800
Subject: Fix TestGateway for TCP services (#35782)

This is broken since we do the port -> nodeport translation twice in the
original code.
---
 tests/integration/pilot/ingress_test.go | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/tests/integration/pilot/ingress_test.go b/tests/integration/pilot/ingress_test.go
index 4e24babf17..f4638b9a3c 100644
--- a/tests/integration/pilot/ingress_test.go
+++ b/tests/integration/pilot/ingress_test.go
@@ -205,14 +205,12 @@ func TestGateway(t *testing.T) {
 						}
 					})
 					t.NewSubTest("tcp").Run(func(t framework.TestContext) {
-						host, port := apps.Ingress.TCPAddress()
 						_ = apps.Ingress.CallWithRetryOrFail(t, echo.CallOptions{
 							Port: &echo.Port{
 								Protocol:    protocol.HTTP,
-								ServicePort: port,
+								ServicePort: 31400,
 							},
-							Address: host,
-							Path:    "/",
+							Path: "/",
 							Headers: map[string][]string{
 								"Host": {"my.domain.example"},
 							},
-- 
2.35.3

